src/signTx.c

```diff
@@ -666,6 +735,12 @@ static void signTx_handleInput_ui_runStep()
 
 	UI_STEP_BEGIN(ctx->ui_step, this_fn);
 
+	UI_STEP(HANDLE_INPUT_STEP_DISPLAY) {
+		char headerText[18];
```

Comment:
I'd bzero this, I don't remember if the Ledger snprintf adds null terminator, and let's add an assert after snprintf about the length of headerText.